### ISSUE TITLE
Recommended way to go after upgrade to SpringBoot3 - attributes 

### ISSUE THREAD

**Issue by ottlinger** [thymeleaf/thymeleaf#920]
After playing around with the latest SpringBoot 3.0-RC1 I realized my application is not working anymore due to:

`Caused by: java.lang.IllegalArgumentException: The 'request','session','servletContext' and 'response' expression utility objects are no longer available by default for template expressions and their use is not recommended. In cases where they are really needed, they should be manually added as context variables.
`

Is there recommended way to integrate these attributes into the views?
Personally I wouldn't want to integrate all of these things manually in all my controller as I have request-specific highlighting in the navigation bar, which applies to all of my app components.

An example for that is:
```
                <li class="nav-item"
                    th:classappend="${(#request.getServletPath() == '/' ||  #request.getServletPath() == '') ? 'active' : ''}">
                    <a aria-current="page" class="nav-link" th:href="@{/}"><i class="fas fa-home ml-2"></i> <span
                            th:text="#{menu.home}"></span></a>
```

I hope that this is the correct project to file this issue - thanks for any advice/hint in advance.

**Comment by danielfernandez** [comment]
These objects are not directly available in templates in Thymeleaf 3.1 for security reasons. The recommended way to make this information available to templates is to add the specific pieces of information that are really needed by the template as context variables (_model attributes_ in Spring).

So in your case, the base case would be your controller code adding the servlet path to the model like e.g.:

```java
   model.addAttribute("servletPath", request.getServletPath();
```

...and then you would use the `servletPath` context variable normally in your template.

However, you mention that you would prefer not to add this in all your controllers.

In such case, in order to avoid doing this in each of the controller methods of your controller class, you can add the parts of the request you need just once by making use of Spring's `@ModelAttribute` annotation. Or even if you want to make these parts of the request available to absolutely _all_ controller methods, then you could create a `@ControllerAdvice` class that registers these `@ModelAttribute` methods for all controllers in your app.

And if you need several pieces of data from your `HttpServletRequest` and you don't want these to be declared by multiple `@ModelAttribute` methods, then you can just group them into a value object with getter methods and create a single `@ModelAttribute`-annotated method returning that value object.

Hope this helps.

**Comment by ottlinger** [comment]
Thanks for the explanation, I'll see how this integrates into my application but I loved the way it was before: very easy to integrate ;)
